x/wasm-storage/keeper/filter.go

```diff
@@ -74,19 +79,29 @@ func calculate[T comparable](reveals []T) ([]bool, bool) {
 //
 // Note: <param: tallyInput> is a rlp encoded and <param:reveals> is JSON serialized.
 func Outliers(filterInput []byte, reveals []RevealBody) ([]bool, bool, error) {
-	var filter filterProp
-	if err := rlp.DecodeBytes(filterInput, &filter); err != nil {
+	var rlpAsList []any
+	if err := rlp.DecodeBytes(filterInput, &rlpAsList); err != nil {
 		return nil, false, err
 	}
+	filteringAlgo, ok := rlpAsList[0].([]uint)
+	if !ok || len(filteringAlgo) != 1 {
+		return nil, false, errors.New("can not RLP decode algo type from filter input")
+	}
```

Comment:
Was confused about these lines. Why did you use RLP decoding here? I changed the code so that it just looks at the first byte. Let me know what you think.